### PR TITLE
fix clanging scales triggering defense drop twice in double

### PR DESCRIPTION
**Pull request by kleeenexfeu**
Addresses https://github.com/rh-hideout/pokeemerald-expansion/issues/1654
EDIT : does need [parental bond PR](https://github.com/rh-hideout/pokeemerald-expansion/pull/1676) x)
Note that I modified the function IsMoveAffectByParentalBond in Buffel's PR, and also renamed it IsMoveParentalBondAffect. Found it more practical to have a shorter prefix will still having the same meaning.
There's also one more edge case that needs fixing.

### CODE REVIEW

File: src/battle_script_commands.c [hunk 1]
```diff
@@ -3486,9 +3486,77 @@ void SetMoveEffect(bool32 primary, u32 certain)
...
+        switch (gBattleMoves[move].target)
+        {
+            case MOVE_TARGET_BOTH:
+                if (CountAliveMonsInBattle(BATTLE_ALIVE_DEF_SIDE) >= 2) // Check for single target
+                    return TRUE;
+                break;
+            case MOVE_TARGET_FOES_AND_ALLY:
+                if (CountAliveMonsInBattle(BATTLE_ALIVE_EXCEPT_ATTACKER) >= 2) // Count mons on both sides; ignore attacker
+                    return TRUE;
+                break;
```
Comment (LOuroboros) [inline src/battle_script_commands.c:3510]:
```suggestion
        case MOVE_TARGET_BOTH:
            if (CountAliveMonsInBattle(BATTLE_ALIVE_DEF_SIDE) >= 2) // Check for single target
                return TRUE;
            break;
        case MOVE_TARGET_FOES_AND_ALLY:
            if (CountAliveMonsInBattle(BATTLE_ALIVE_EXCEPT_ATTACKER) >= 2) // Count mons on both sides; ignore attacker
                return TRUE;
            break;
```

File: src/pokemon.c [hunk 2]
```diff
@@ -3057,8 +3057,14 @@ u8 CountAliveMonsInBattle(u8 caseId)
                 retVal++;
         }
         break;
+	case BATTLE_ALIVE_EXCEPT_ATTACKER:
```
Comment (LOuroboros) [inline src/pokemon.c:3060]: Spaces instead of tab.

File: src/pokemon.c [hunk 3]
```diff
@@ -3057,8 +3057,14 @@ u8 CountAliveMonsInBattle(u8 caseId)
...
+        for (i = 0; i < MAX_BATTLERS_COUNT; i++)
+        {
+            if (i != gBattlerAttacker && !(gAbsentBattlerFlags & gBitTable[i]))
+                retVal++;
+        }
+		break;
```
Comment (LOuroboros) [inline src/pokemon.c:3066]: Spaces instead of tab.

**Review by LOuroboros — CHANGES_REQUESTED** [review]

**Comment by AsparagusEduardo** [comment]
Closing this for now, as the required PR is in review purgatory and after such a long time it'll be better to tackle this in a new PR.

**Pull request closed without merging**